src/software/ai/navigator/navigator.cpp

```diff
@@ -2,303 +2,285 @@
-                Point(10, world.field().enemyDefenseArea().negXNegYCorner().y()));
-            rectangle.expand(Util::DynamicParameters->getNavigatorConfig()
-                                 ->RobotObstacleInflationFactor()
-                                 ->value() *
-                             ROBOT_MAX_RADIUS_METERS);
-            return Obstacle(rectangle);
-        case AvoidArea::INFLATED_ENEMY_DEFENSE_AREA:
-            rectangle = world.field().enemyDefenseArea();
-            rectangle.expand(Util::DynamicParameters->getNavigatorConfig()
-                                     ->RobotObstacleInflationFactor()
-                                     ->value() *
-                                 ROBOT_MAX_RADIUS_METERS +
-                             0.3);  // 0.3 is by definition what inflated means
-            return Obstacle(rectangle);
-        case AvoidArea::CENTER_CIRCLE:
-            return Obstacle::createCircleObstacle(
-                world.field().centerPoint(), world.field().centerCircleRadius(),
-                Util::DynamicParameters->getNavigatorConfig()
-                    ->RobotObstacleInflationFactor()
-                    ->value());
-        case AvoidArea::HALF_METER_AROUND_BALL:
-            return Obstacle::createCircleObstacle(
-                world.ball().position(), 0.5,  // 0.5 represents half a metre radius
-                Util::DynamicParameters->getNavigatorConfig()
-                    ->RobotObstacleInflationFactor()
-                    ->value());
-        case AvoidArea::BALL:
-            return Obstacle::createCircularBallObstacle(world.ball(), 0.06);
-        case AvoidArea::ENEMY_HALF:
-            rectangle = Rectangle({0, world.field().totalYLength() / 2},
-                                  world.field().enemyCornerNeg() -
-                                      Point(0, world.field().boundaryYLength()));
-            rectangle.expand(Util::DynamicParameters->getNavigatorConfig()
-                                 ->RobotObstacleInflationFactor()
-                                 ->value() *
-                             ROBOT_MAX_RADIUS_METERS);
-            return Obstacle(rectangle);
-        case AvoidArea::FRIENDLY_HALF:
-            rectangle = Rectangle({0, world.field().totalYLength() / 2},
-                                  world.field().friendlyCornerNeg() -
-                                      Point(0, world.field().boundaryYLength()));
-            rectangle.expand(Util::DynamicParameters->getNavigatorConfig()
-                                 ->RobotObstacleInflationFactor()
-                                 ->value() *
-                             ROBOT_MAX_RADIUS_METERS);
-            return Obstacle(rectangle);
-        default:
-            LOG(WARNING) << "Could not convert AvoidArea " << (int)avoid_area
-                         << " to obstacle";
+        auto path      = robot_id_to_path[intent.getRobotId()];
```

Comment:
What happens if we lookup an robot ID that doesn't exist in the map? In this case I think something has pretty obviously gone badly wrong, so if we throw an exception that can be caught higher up that's all good, but just want to check this won't segfault.

Reply:
it'll actually return a std::nullopt. We want to handle that case explicitly and return a warning, right?

Reply:
Yeah, we should log a warning and not use the `std::nullopt` value below

Reply:
Do we throw an exception?

Reply:
Well throwing an exception now

Reply:
Sorry, a bit confused here:
- `map::operator[]` does not return a `std::nullopt`, and in fact gives no error if an element was not in the map (it actually adds one if it wasn't already there)
- `map::at` will just throw a `out_of_range` exception 

Just to be clear, what were you talking about when you said "it'll actually return a std::nullopt"?

Now regarding just throwing an exception, it's not very helpful if at the top level we just catch a `out_of_range` exception, since we will have no idea where it came from. We should at minimum add a descriptive message to the exception before just throwing it up, unless there's some exception handling in the Navigator that I'm missing?

Reply:
[] would return a nullopt if it's not found in the map. Ok I can make it more descriptive

Reply:
Done

Reply:
This looks good to me, just leaving the comment open while we clarify offline about what `[]` returns

Reply:
Ah, clarified offline. Basically we're mapping to a `std::optional<path>` when the `[]` operator doesn't find an element, it inserts one created using the default constructor and returns that instead. Since we were inserting an optional type, the default constructor results in an `std::nullopt`. It's definitely better to be clear here though!